Large update to contributing doc

kwirkykat opened the pull request:
I updated the contributor instructions document to include sections on how to submit an issue and how to review a pull request with Reviewable.

I also updated the other sections on fixing an issue, submitting a pull request, writing documentation, and submitting new resources/resource modules.

Probably easier to view the rendered markdown file in my fork [here](https://github.com/kwirkykat/DscResources/blob/UpdateContributingDoc/CONTRIBUTING.md).

This update purposefully does not include more specific instructions surrounding writing tests as both a new testing document and updates to the testing templates are currently in progress.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/dscresources/182)
<!-- Reviewable:end -->

mbreakey3 commented:
Review status: 0 of 24 files reviewed at latest revision, 3 unresolved discussions.

---

*[CONTRIBUTING.md, line 14 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQarpyisF7ORDFnP9k5:-KQarpyisF7ORDFnP9k6:b-2r2d6m) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L14)):*
> ```Markdown
> * [Review pull requests](#reviewing-pull-requests) 
> 
> If you're just starting out with GitHub, start by reading our [guide to getting started with GitHub](GettingStartedWithGitHub.md).
> ```

Right now this links to the doc in your fork - will this change when it gets merged?

---

*[CONTRIBUTING.md, line 37 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQastkgm4z7Z0XRWIb3:-KQastkgm4z7Z0XRWIb4:bwsoelr) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L37)):*
> ```Markdown
> | <ul><li> Common tests </li><li> Meta-fixers </li></ul> | [DscResource.Tests](https://github.com/PowerShell/DscResource.Tests)
> | <ul><li> Specific resource </li><li> Resource proposal </li></ul> | The repository of the resource module that contains/should contain the resource. |
> | <ul><li> Specific resource module </li></ul> | The repository of that resource module |
> ```

Make this a little more specific like: Any bugs, enhancements, etc. to a specific resource or module

---

*[CONTRIBUTING.md, line 48 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQatTR6NCaqQa0F70CW:-KQatTR6NCaqQa0F70CX:b37f2v5) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L48)):*
> ```Markdown
> ```
> 
> All DSC resource modules are also listed as submodules of this repository under the [xDscResources](xDscResources) and [DscResources](DscResources) folders.
> ```

This link to the DscResources isn't working currently either

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: 0 of 24 files reviewed at latest revision, 3 unresolved discussions.

---

*[CONTRIBUTING.md, line 14 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQarpyisF7ORDFnP9k5:-KQb5J7Y3YfWGXx_hJS0:b-p5t8fh) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L14)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Right now this links to the doc in your fork - will this change when it gets merged?
</details>
Yes. It's tied to whatever is in the local branch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@mbreakey3 Link should be fixed now. Updated the descriptions in the table as well.

mbreakey3 commented:
Review status: 0 of 24 files reviewed at latest revision, 28 unresolved discussions.

---

*[CONTRIBUTING.md, line 156 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbYfFT0nMVIvLri7cL:-KQbYfFT0nMVIvLri7cM:bn3nt11) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L156)):*
> ```Markdown
> 		* Unit  
> 			* MyResource.Tests.ps1  
> 		* Integrations  
> ```

'Integration'

---

*[CONTRIBUTING.md, line 182 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbZ2CvmewmlEA7eMok:-KQbZ2CwLhFQ5_PSyPWt:b-czufdy) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L182)):*
> ```Markdown
> ...
> ```
> If a change is related to a specific resource, please add create sub-bullets with your changes under a bullet with the resource name.  
> ```

'please add sub-bullets' - get rid of the create

---

*[CONTRIBUTING.md, line 203 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbZItgyF_7Ur10jx-3:-KQbZItgyF_7Ur10jx-4:bcnrcxl) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L203)):*
> ```Markdown
> 
> ### Submit a Pull Request
> A [pull request](https://help.github.com/articles/using-pull-requests/)(PR) allows you to submit the changes you made in your fork to the official repository.
> ```

space between link and '(PR)'

---

*[CONTRIBUTING.md, line 233 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbk8VRuBLTYGaxw_BT:-KQbk8VRuBLTYGaxw_BU:b50u2yv) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L233)):*
> ```Markdown
> Simply putting the issue number that the PR fixes is not acceptable.
> If you PR deals with *one* specific resource, please prefix the title with the resource name followed by a colon.
> If your PR does fix an issue please do still include "(Fixes #issue number)" in the title.
> ```

'If your PR fixes an issue, please include.."

---

*[CONTRIBUTING.md, line 234 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbkYRFtyN3thyHeyeD:-KQbkYRFtyN3thyHeyeE:bkkkcgl) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L234)):*
> ```Markdown
> If you PR deals with *one* specific resource, please prefix the title with the resource name followed by a colon.
> If your PR does fix an issue please do still include "(Fixes #issue number)" in the title.
> For example, if a PR fixes issues number 11 and 16 which adds the Ensure parameter to the xSample resource:
> ```

'...to the xSample resource, the title should be something like: '

---

*[CONTRIBUTING.md, line 258 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbslMXXqU79UXkt6qk:-KQbslMXXqU79UXkt6ql:bbf3rav) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L258)):*
> ```Markdown
> The DSC Resource Kit uses [AppVeyor](http://www.appveyor.com/) as a continuous integration (CI) system.
> 
> After submitting your pull request, AppVeyor will run a suite of tests and automatically update the status of the pull request and providing at-a-glance feedback about whether your changes are passing tests or not.
> ```

'and provide at-a-glance'

---

*[CONTRIBUTING.md, line 265 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbt6sChQW6b7wjdBql:-KQbt6sChQW6b7wjdBqm:b-ikjng8) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L265)):*
> ```Markdown
> 
> A maintainer **will not** merge your pull request if these tests are failing, even if they have nothing to do with your changes.
> If test failures are occurring that do not relate to the changes you made, you will have to submit another PR with fixes or that issue or wait until someone else does.
> ```

'on that issue'

---

*[CONTRIBUTING.md, line 300 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbuIMWTP8x_IJnipKE:-KQbuIMXfAlM5FlTz84x:beyilg5) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L300)):*
> ```Markdown
> DSCResource.Tests should be in the .gitignore file so that git will automatically ignore this folder.
> If DSCResource.Tests is not in the .gitignore file, please add it.
> If there is no .gitignore file for your module, instructions on how to add one are available in our [getting started with GitHub](GettingStartedWithGitHub.md) instructions.
> ```

Could we add a table of contents to the Getting started with GitHub file? Then direct these links to the corresponding content in the file?

---

*[CONTRIBUTING.md, line 309 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbudiPqHcCF3aI5DI6:-KQbudiQoAS7FpDkK0pG:bz556el) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L309)):*
> ```Markdown
> 
> ### Get your Code Reviewed
> Anyone other than you can *review* your code, but only maintainers may *merge* your code.
> ```

'but only maintainers can *merge*...'

---

*[CONTRIBUTING.md, line 328 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbvE5umsIIzF_FULF5:-KQbvE5umsIIzF_FULF6:b-z2d2rk) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L328)):*
> ```Markdown
> - Fundamentally changing an existing functionality of a resource
> 
> Once a PR is ready with the breaking change please include the following:
> ```

need a line break between this one and the following one (starting with '1.')

---

*[CONTRIBUTING.md, line 346 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbwN0UIkiw9b4-gyvR:-KQbwN0VA2uSlMnxydP2:bwiocah) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L346)):*
> ```Markdown
> If someone is already working on a similar module, please leave a comment on the open issue or add a GitHub reaction to the top comment to **express your interest**. You can also offer help and use the issue to coordinate your efforts.
> 
> If none of the existing module proposals or submissions are related to one you want, [open a new issue](https://github.com/PowerShell/DscResources/issues) with the following information:
> ```

maybe change this to 'open a new issue [in this repository](...) 

---

*[CONTRIBUTING.md, line 352 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbwtJjSUynPLyBvEB4:-KQbwtJjSUynPLyBvEB5:b-g3juke) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L352)):*
> ```Markdown
> * What resources you would like to be included in the new resource module?
> * Will your module include [MOF-based resources](https://technet.microsoft.com/en-us/library/dn956964.aspx) (compatible with PS/WMF 4.0 and 5.0+) or [class-based resources](https://technet.microsoft.com/en-us/library/dn948461.aspx) (only compatible with PS/WMF 5.0+)
> * Are or will you be working on this module or would you like someone else to pick it up?
> ```

'* Are you or will you...'

---

*[CONTRIBUTING.md, line 368 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxXqrsXLJDQEHBAfv:-KQbxXqrsXLJDQEHBAfw:b-auwu37) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L368)):*
> ```Markdown
> 		    * SampleResource.Tests.ps1  
> 		  * Integration
> 		    * SampleResource.Integration.Tests.ps1
> ```

is including a config file no longer the norm for Integration tests?

---

*[CONTRIBUTING.md, line 369 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxEtFXiCS9xAu_8-2:-KQbxEtFXiCS9xAu_8-3:b-ms2zr6) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L369)):*
> ```Markdown
> 		  * Integration
> 		    * SampleResource.Integration.Tests.ps1
> 	  * Examples
> ```

This line needs to be tabbed left once.

---

*[CONTRIBUTING.md, line 370 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbx8fPog9DXSqTKN12:-KQbx8fPog9DXSqTKN13:bfq2g2h) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L370)):*
> ```Markdown
> 		    * SampleResource.Integration.Tests.ps1
> 	  * Examples
> 	    * SampleResource.Example.ps1 
> ```

this line under 'Examples' needs to be tabbed in a couple times

---

*[CONTRIBUTING.md, line 371 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxKNnkGK2YJ8kqo67:-KQbxKNoB15LXdJMGApP:b-4r8gvr) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L371)):*
> ```Markdown
> 	  * Examples
> 	    * SampleResource.Example.ps1 
> 	  * Module manifest (e.g. SampleResourceModuleDsc.psd1)
> ```

this line and the two below need to be tabbed left one

---

*[CONTRIBUTING.md, line 378 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQby1KZRr9g539Dlsrv:-KQby1KZRr9g539Dlsrw:b-7ugjc4) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L378)):*
> ```Markdown
> * Run the common tests located in [DSCResource.Tests](https://github.com/PowerShell/DscResource.Tests) which will be automatically installed into the root folder of your module when your tests are run if your module is accepted in the DSC Resource Kit. 
> 
> All new resource modules should follow the DSC Resource Kit [High Quality guidelines](HighQualityModuleGuidelines.md)
> ```

add a period at the end of this line.

---

*[CONTRIBUTING.md, line 385 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbyOXWlXfrNIoojFM0:-KQbyOXXpyUnvMLRqGrM:btawgm5) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L385)):*
> ```Markdown
> ## Writing Documentation
> One of the easiest ways to contribute to a PowerShell project is to write and edit documentation. 
> All of our documentation hosted on GitHub is written using [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/) in the style of [this example](DscResourcesExampleHelp.md).
> ```

Is there a better link for the GitHub Flavored Markdown? Also the 'this example' link isn't working from your fork at least

---

*[CONTRIBUTING.md, line 389 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbyjVeb5lp8VWSytcs:-KQbyjVfV2O-SqJFZJGV:b-668t56) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L389)):*
> ```Markdown
> To [edit an existing file](https://help.github.com/articles/editing-files-in-another-user-s-repository/), simply navigate to it in GitHub and click the "Edit" button. 
> GitHub will *automatically* create your own fork of the repository where you can make your changes. 
> Once you're finished, commit your edits and [submit a pull request](#submit-a-pull-request) to get your changes merged into the official repository. 
> ```

'Once you're finished, commit your changes..'

---

*[CONTRIBUTING.md, line 412 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbzmpSq3NaPl29MzJ-:-KQbzmpSq3NaPl29MzJ0:b-uw6g5z) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L412)):*
> ```Markdown
> **Pull requests should not be reviewed while tests from AppVeyor are failing.**
> 
> All modules in the DSC Resource Kit should be linked to [Reviewable](reviewable.io), a code review tool.
> ```

the Reviewable link doesn't go to Reviewable

---

*[CONTRIBUTING.md, line 418 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc--pkOw7GR-BG_44g:-KQc--pkOw7GR-BG_44h:b-klroul) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L418)):*
> ```Markdown
> It will take you to a code review of all the changes in that pull request.
> 
> If the purple Reviewable button does not appear, you can also go [here](https://reviewable.io/reviews) and paste the URL of the pull request you would like to review into this box:  
> ```

This link to reviewable, however, does work.
Also change to: 'like to review into this box at the bottom of the page:'

---

*[CONTRIBUTING.md, line 432 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc-RagkT9jM38Zdn9U:-KQc-RagkT9jM38Zdn9V:b-mu3qlu) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L432)):*
> ```Markdown
> * Does the code follow the DSC Resource Kit [Style Guidelines](StyleGuidelines.md) and [Best Practices](BestPractices.md)?
> * Has the author included test coverage for their changes?
> * Has the author updated the Unreleased section of the README with their changes?
> ```

Add: 
* Is the code sufficiently commented and easy to read?

---

*[CONTRIBUTING.md, line 442 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc01o-9GQiJzavUQIT:-KQc01o0wPwGkf7ZlYvj:b502p3o) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L442)):*
> ```Markdown
> ![ReviewableDiscussionSatisfied](Images/ReviewableDiscussionSatisfied.png)
> 
> To mark an entire file as reviewed, click the little eye button next to the file name at the top of the file so that it is green.  
> ```

'so that it turns green.'

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KQc0FIGW7h_92H-X_mX:b58pbnw) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
> ```Markdown
> ![ReviewableFilePicker](Images/ReviewableFilePicker.png) 
> 
> Nothing you do in Reviewable will be pushed to the public or the pull request until you hit the green Publish button at the top:  
> ```

'until you hit the green Publish button at the top or the little publish button in the corner of every comment - but please publish all or a large chunk of comments at once so that other participants in the review don't get bombarded with notifications '

---

*[CONTRIBUTING.md, line 455 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0y8NNHmJp7DtXNP6:-KQc0y8NNHmJp7DtXNP7:b-57a684) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L455)):*
> ```Markdown
> 
> Nothing you do in Reviewable will be pushed to the public or the pull request until you hit the green Publish button at the top:  
> ![ReviewablePublishButton](Images/ReviewablePublishButton.png)
> ```

Maybe add a closing statement as well, like: Thank you for taking the time to read through our guidelines! I know it's a lot of information, but once you get started it's not as overwhelming as it seems. Feel free to reach out to me (@kwirkykat) or Mariah (@mbreakey3) if you have any questions or concerns

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 1 of 1 files at r3.
Review status: 1 of 24 files reviewed at latest revision, 32 unresolved discussions.

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KQkX_IHlj7cDxz7Ch6E:b-88mj22) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'until you hit the green Publish button at the top or the little publish button in the corner of every comment - but please publish all or a large chunk of comments at once so that other participants in the review don't get bombarded with notifications '
</details>
Agree, with a little twist. For the beginners of reviewable, write something like this (please do make the text better ;)
 "comments are saved as a draft so that you can continue commenting on all files in the PR and publish everything at once. And that is the preferred way to publish all at once using the "big green Publish button". And one should try to avoid using the "little publish button" which will create a comment on the PR on GitHub for each comment, subsequently will send an e-mail to all who watches for each comment." 

(this one is for beginners to)
Also maybe add something like "When your press Publish, Reviewable will gather all your saved draft comments and add them as a single comment to the pull request on GitHub." 

---

*[CONTRIBUTING.md, line 55 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkRH-XLRcNFFT6vk87:-KQkRH-XLRcNFFT6vk88:br3ltbr) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L55)):*
> ```Markdown
> 
> **Ensure that the issue you are about to file is not already open.**
> If you cannot find an issue that matches the one you are about to file, click the New Issue button on the right.
> ```

Say something about that if you do find an issue that matches, you could use GitHub reactions to upvote?

---

*[CONTRIBUTING.md, line 73 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkGnld8_nzt5XZjgTt:-KQkGnld8_nzt5XZjgTu:b-spyjnr) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L73)):*
>If GitHub automatically chooses the wrong issue/pull request (e.g. it chooses pull request #18 instead of issue #18)

Can an issue and a pull request have the same number? Have that happened? Seems that numbering for both issues and PR's are the same number series and can't conflict? But I might be wrong.

Maybe also add that you can reference issues and pull request in other repos by specifying something like `PowerShell/OtherModuleDsc#27`?

---

*[CONTRIBUTING.md, line 235 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkLuCdgQYJzqqII1-C:-KQkLuCedVQrJjGHiuRg:b-qvfmkw) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L235)):*
> ```Markdown
> If your PR does fix an issue please do still include "(Fixes #issue number)" in the title.
> For example, if a PR fixes issues number 11 and 16 which adds the Ensure parameter to the xSample resource:
> "xSample: Added Ensure parameter (Fixes #11, #16)".
> ```

You should make a reference here to the breaking changes section too. Title should then begin with `BREAKING CHANGE: `

Maybe also add that if one does get the title wrong, it is easily fixed by editing the title in the PR?

---

*[CONTRIBUTING.md, line 249 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkN8Xf0jDNFfSag6br:-KQkN8XgVX3_LEYszWS9:b-i2zg02) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L249)):*
> ```Markdown
> 
> Once you sign the CLA, the Microsoft CLA bot will automatically update your PR:
> ![GitHubCLASigned.png](Images/GitHubCLASigned.png)
> ```

Just making sure. In the previous image you redacted the username, but not in this one. 

---

*[CONTRIBUTING.md, line 329 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkPpbox1PRj3CpV5TZ:-KQkPpbox1PRj3CpV5T_:btwnz0r) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L329)):*
> ```Markdown
> 
> Once a PR is ready with the breaking change please include the following:
> 1. At least one of the bullet points in your addition to the updated release notes starts with 'BREAKING CHANGE:'    
> ```

This numbered list should start on a new paragraph (currently 1. continues on the previous row)

---

*[CONTRIBUTING.md, line 387 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkSFUfWuX5b7asLnSR:-KQkSFUfWuX5b7asLnSS:b9extt4) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L387)):*
> ```Markdown
> All of our documentation hosted on GitHub is written using [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/) in the style of [this example](DscResourcesExampleHelp.md).
> 
> To [edit an existing file](https://help.github.com/articles/editing-files-in-another-user-s-repository/), simply navigate to it in GitHub and click the "Edit" button. 
> ```

Maybe move this paragraph under the numbered list? (So one first read that one should first check/submit issues and then start edit.)

---

*[CONTRIBUTING.md, line 410 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkTF77V3y4TrsS5tto:-KQkTF77V3y4TrsS5ttp:b-32pdp4) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L410)):*
> ```Markdown
> All pull requests across all modules in the DSC Resource Kit that are open for review can be seen in the Needs Review column on our [dashboard](https://waffle.io/powershell/dscresources/join).
> 
> **Pull requests should not be reviewed while tests from AppVeyor are failing.**
> ```

Maybe add here the if tests are failing and you don't know why, tag a maintainer or ask the community for help? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: 1 of 24 files reviewed at latest revision, 34 unresolved discussions.

---

*[CONTRIBUTING.md, line 37 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQastkgm4z7Z0XRWIb3:-KQvpAYVzG1RDOwpxlWm:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L37)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Make this a little more specific like: Any bugs, enhancements, etc. to a specific resource or module
</details>
Done.

---

*[CONTRIBUTING.md, line 48 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQatTR6NCaqQa0F70CW:-KQvpB8ygpzF22sEAnpR:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L48)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> This link to the DscResources isn't working currently either
</details>
Done.

---

*[CONTRIBUTING.md, line 156 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbYfFT0nMVIvLri7cL:-KQvsILmCJtGyz-RApAT:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L156)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'Integration'
</details>
Done.

---

*[CONTRIBUTING.md, line 182 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbZ2CvmewmlEA7eMok:-KQvsRfYcV7fpj5eEwRJ:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L182)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'please add sub-bullets' - get rid of the create
</details>
Done.

---

*[CONTRIBUTING.md, line 203 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbZItgyF_7Ur10jx-3:-KQvs_WyxjuNrsgoE_Ac:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L203)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> space between link and '(PR)'
</details>
Done.

---

*[CONTRIBUTING.md, line 233 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbk8VRuBLTYGaxw_BT:-KQvslyv_2nPkqLK2lHr:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L233)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'If your PR fixes an issue, please include.."
</details>
Done.

---

*[CONTRIBUTING.md, line 234 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbkYRFtyN3thyHeyeD:-KQvsymZ1-8xROM6r_2S:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L234)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> '...to the xSample resource, the title should be something like: '
</details>
Done.

---

*[CONTRIBUTING.md, line 258 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbslMXXqU79UXkt6qk:-KQvubi4eq3C_E8MQ4sU:bpjod68) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L258)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'and provide at-a-glance'
</details>
Switched around the wording.

---

*[CONTRIBUTING.md, line 265 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbt6sChQW6b7wjdBql:-KQvuto5DaWB22A22izy:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L265)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'on that issue'
</details>
Done.

---

*[CONTRIBUTING.md, line 300 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbuIMWTP8x_IJnipKE:-KQvuw_nBYHLkqfrmJG3:b3reja1) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L300)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Could we add a table of contents to the Getting started with GitHub file? Then direct these links to the corresponding content in the file?
</details>
Yes, that file needs to be re-done as well but not as part of this PR. Can you open an issue for this?

---

*[CONTRIBUTING.md, line 309 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbudiPqHcCF3aI5DI6:-KQvv7YCeGUrjIV53NrI:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L309)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'but only maintainers can *merge*...'
</details>
Done.

---

*[CONTRIBUTING.md, line 328 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbvE5umsIIzF_FULF5:-KQvvBuny2meJbUrOs9L:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L328)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> need a line break between this one and the following one (starting with '1.')
</details>
Done.

---

*[CONTRIBUTING.md, line 346 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbwN0UIkiw9b4-gyvR:-KQvviQsMjXwPnUkngmP:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L346)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> maybe change this to 'open a new issue [in this repository](...) 
</details>
Done.

---

*[CONTRIBUTING.md, line 352 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbwtJjSUynPLyBvEB4:-KQvvmbd28C46I3zOSD2:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L352)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> '* Are you or will you...'
</details>
Done.

---

*[CONTRIBUTING.md, line 368 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxXqrsXLJDQEHBAfv:-KQvvrSv129PfbZqdDo8:b-pcsqcu) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L368)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> is including a config file no longer the norm for Integration tests?
</details>
In xPSDSC we don't have separate config files...
It creates a lot of extra files to have each config as its own file if we want to test more than one configuration within one integration test file.
I'm not going to include it in this structure as I would consider it optional and I would rather leave it open-ended in this document.

---

*[CONTRIBUTING.md, line 369 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxEtFXiCS9xAu_8-2:-KQvwslpA-kCiXTFXq5A:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L369)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> This line needs to be tabbed left once.
</details>
Done.

---

*[CONTRIBUTING.md, line 370 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbx8fPog9DXSqTKN12:-KQvwtyBVDP9KsHBsgOj:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L370)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> this line under 'Examples' needs to be tabbed in a couple times
</details>
Done.

---

*[CONTRIBUTING.md, line 371 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbxKNnkGK2YJ8kqo67:-KQvwvqV00qqA751dW1-:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L371)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> this line and the two below need to be tabbed left one
</details>
Done.

---

*[CONTRIBUTING.md, line 378 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQby1KZRr9g539Dlsrv:-KQvx16SSi9FvXbhwbVY:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L378)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> add a period at the end of this line.
</details>
Done.

---

*[CONTRIBUTING.md, line 385 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbyOXWlXfrNIoojFM0:-KQvxdJHkryyVCZ4g8OO:b-1apxwl) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L385)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Is there a better link for the GitHub Flavored Markdown? Also the 'this example' link isn't working from your fork at least
</details>
Updated the link. I'm not sure what the example link was originally supposed to point to.

---

*[CONTRIBUTING.md, line 389 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbyjVeb5lp8VWSytcs:-KQw-CPULSAMAAYATar4:b88ghcb) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L389)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'Once you're finished, commit your changes..'
</details>
Rearranged this whole section since it was a bit repetitive.

---

*[CONTRIBUTING.md, line 412 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQbzmpSq3NaPl29MzJ-:-KQw-NLdgiQdZIsIcMH_:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L412)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> the Reviewable link doesn't go to Reviewable
</details>
Done.

---

*[CONTRIBUTING.md, line 418 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc--pkOw7GR-BG_44g:-KQw-Zco_N7n7NHXq8n1:b-x2nhmg) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L418)):*
> at the bottom of the page:
If you have no other reviews open, it's not really at the bottom of the page.
I added 'towards the bottom of the page' at least...

---

*[CONTRIBUTING.md, line 432 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc-RagkT9jM38Zdn9U:-KQw-rV9zKJtokplp605:b-jo8wzm) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L432)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Add: 
> * Is the code sufficiently commented and easy to read?
</details>
I'm actually not a fan of comments in code. I think it generally means that the code was not written clearly enough to begin with. Anyways, the point is that that's a more maintainer's-opinion point rather than a general rule. But I did add that the code should be easy to read and every function should have help-comments because those are very useful and required for the resource kit.

---

*[CONTRIBUTING.md, line 442 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc01o-9GQiJzavUQIT:-KQw1b04CHAU60a0Vwmj:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L442)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'so that it turns green.'
</details>
Done.

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KQw5x0i3qqUhbv_bMqQ:b4his1u) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Agree, with a little twist. For the beginners of reviewable, write something like this (please do make the text better ;)
>  "comments are saved as a draft so that you can continue commenting on all files in the PR and publish everything at once. And that is the preferred way to publish all at once using the "big green Publish button". And one should try to avoid using the "little publish button" which will create a comment on the PR on GitHub for each comment, subsequently will send an e-mail to all who watches for each comment." 
> 
> (this one is for beginners to)
> Also maybe add something like "When your press Publish, Reviewable will gather all your saved draft comments and add them as a single comment to the pull request on GitHub." 
</details>
I rearranged this section as well and split it into sub-sections. I also added a little more detail about the drafts and publishing.

---

*[CONTRIBUTING.md, line 455 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0y8NNHmJp7DtXNP6:-KQw6nG6fyFnSJjx1TRo:b-rxuwdi) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L455)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Maybe add a closing statement as well, like: Thank you for taking the time to read through our guidelines! I know it's a lot of information, but once you get started it's not as overwhelming as it seems. Feel free to reach out to me (@kwirkykat) or Mariah (@mbreakey3) if you have any questions or concerns
</details>
I'm not going to add a sign-off, but I did add the line about contacted you or me in the first section at the top.

---

*[CONTRIBUTING.md, line 55 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkRH-XLRcNFFT6vk87:-KQvpBd2vR95IcgDuVYe:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L55)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Say something about that if you do find an issue that matches, you could use GitHub reactions to upvote?
</details>
Done.

---

*[CONTRIBUTING.md, line 73 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkGnld8_nzt5XZjgTt:-KQvqmAVNFGQ-TuqORLs:b-5y3wld) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L73)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> >If GitHub automatically chooses the wrong issue/pull request (e.g. it chooses pull request #18 instead of issue #18)
> 
> Can an issue and a pull request have the same number? Have that happened? Seems that numbering for both issues and PR's are the same number series and can't conflict? But I might be wrong.
> 
> Maybe also add that you can reference issues and pull request in other repos by specifying something like `PowerShell/OtherModuleDsc#27`?
> 
</details>
I thought I saw this problem arise before, but maybe it was just that the author had the wrong number.
Will remove the comment for now, and we can always add it back if the problem arises again.

---

*[CONTRIBUTING.md, line 235 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkLuCdgQYJzqqII1-C:-KQvtZQ2y_onG0jkiQAK:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L235)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> You should make a reference here to the breaking changes section too. Title should then begin with `BREAKING CHANGE: `
> 
> Maybe also add that if one does get the title wrong, it is easily fixed by editing the title in the PR?
</details>
Done.

---

*[CONTRIBUTING.md, line 249 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkN8Xf0jDNFfSag6br:-KQw762-k13n3Lntn7iW:b-ksj9c9) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L249)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Just making sure. In the previous image you redacted the username, but not in this one. 
</details>
Thanks for catching this. I fixed it. Unfortunately, GitHub seems to have cached the image on my machine, so I can't see the fixed image in the document, but you can see it has been corrected in this commit: https://github.com/kwirkykat/DscResources/commit/0f025aefbc5af710cfe75e109a8cd9cd1b9ce594

---

*[CONTRIBUTING.md, line 329 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkPpbox1PRj3CpV5TZ:-KQvvCLkLXcJZfYeMxke:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L329)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> This numbered list should start on a new paragraph (currently 1. continues on the previous row)
</details>
Done.

---

*[CONTRIBUTING.md, line 387 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkSFUfWuX5b7asLnSR:-KQw-3FCev49YPjFpkHg:b88ghcb) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L387)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Maybe move this paragraph under the numbered list? (So one first read that one should first check/submit issues and then start edit.)
</details>
Rearranged this whole section since it was a bit repetitive.

---

*[CONTRIBUTING.md, line 410 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkTF77V3y4TrsS5tto:-KQw6DaVBScYK2X1O7i5:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L410)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Maybe add here the if tests are failing and you don't know why, tag a maintainer or ask the community for help? 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 0 of 24 files reviewed at latest revision, 33 unresolved discussions.

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KQzMdCOKCmuxQXnpWgA:bp0t5ol) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> I rearranged this section as well and split it into sub-sections. I also added a little more detail about the drafts and publishing.
</details>
Much better! :SMILE: LGTM

---

*[CONTRIBUTING.md, line 55 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkRH-XLRcNFFT6vk87:-KQzMpa1DocqFGMvWGBi:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L55)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Done.
</details>
LGTM

---

*[CONTRIBUTING.md, line 73 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkGnld8_nzt5XZjgTt:-KQzN-ZdqaPj7bqnhlSH:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L73)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> I thought I saw this problem arise before, but maybe it was just that the author had the wrong number.
> Will remove the comment for now, and we can always add it back if the problem arises again.
</details>
LGTM

---

*[CONTRIBUTING.md, line 235 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkLuCdgQYJzqqII1-C:-KQzN944VRLP_yW8Cofc:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L235)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Done.
</details>
LGTM

---

*[CONTRIBUTING.md, line 249 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkN8Xf0jDNFfSag6br:-KQzNC6Vs_Es5LsuX9ry:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L249)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Thanks for catching this. I fixed it. Unfortunately, GitHub seems to have cached the image on my machine, so I can't see the fixed image in the document, but you can see it has been corrected in this commit: https://github.com/kwirkykat/DscResources/commit/0f025aefbc5af710cfe75e109a8cd9cd1b9ce594
</details>
LGTM

---

*[CONTRIBUTING.md, line 329 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkPpbox1PRj3CpV5TZ:-KQzNVOyUfEqdSRLhwqI:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L329)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Done.
</details>
LGTM

---

*[CONTRIBUTING.md, line 387 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkSFUfWuX5b7asLnSR:-KQzNg4khqvDLcKPmxr3:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L387)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Rearranged this whole section since it was a bit repetitive.
</details>
LGTM

---

*[CONTRIBUTING.md, line 410 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQkTF77V3y4TrsS5tto:-KQzNl3gENx08ONWHfGo:bb6ywmx) ([raw file](https://github.com/powershell/dscresources/blob/2987a3b5380465ea865284b56156ab5c1b89a689/CONTRIBUTING.md#L410)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Done.
</details>
LGTM

---

*[CONTRIBUTING.md, line 458 \[r4\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQzLK3fqHE_oeVxowik:-KQzLK3g7TYnjK-lgdDA:b-glc33o) ([raw file](https://github.com/powershell/dscresources/blob/0f025aefbc5af710cfe75e109a8cd9cd1b9ce594/CONTRIBUTING.md#L458)):*
In the [Maintainers.md](https://github.com/PowerShell/DscResources/blob/master/Maintainers.md) it says 
> while reviewing PRs, leave your comment even if everything looks good - simple "Looks good to me" or "LGTM" will suffice, so that we know someone has already taken a look at it

But in this text says you should resolve the comments. These texts need to be "aligned". :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Review status: 0 of 24 files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KR0SZ_lZ1Y5gI-KS5Dk:b-2xs4ut) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Much better! :SMILE: LGTM
</details>
I think it does still publish everything though when click the little button on the comment...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KR5h6mJBcZ0jGMNVJWz:bs0aid6) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> I think it does still publish everything though when click the little button on the comment...
</details>
Test comment - please ignore :wink: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: 0 of 24 files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 454 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQc0FIGW7h_92H-X_mW:-KR5il19_k0YCKpmPeXD:bs51xkm) ([raw file](https://github.com/powershell/dscresources/blob/c3891f5bcdf19e149dfc97e86d2ad8b19007cf36/CONTRIBUTING.md#L454)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Test comment - please ignore :wink: 
</details>
Nope. Tried it. Reviewable gave me a warning and then published only this comment and not the other one.

---

*[CONTRIBUTING.md, line 458 \[r4\]](https://reviewable.io:443/reviews/powershell/dscresources/182#-KQzLK3fqHE_oeVxowik:-KR5h86o2U8rAebJh2PC:b-qts2bj) ([raw file](https://github.com/powershell/dscresources/blob/0f025aefbc5af710cfe75e109a8cd9cd1b9ce594/CONTRIBUTING.md#L458)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> In the [Maintainers.md](https://github.com/PowerShell/DscResources/blob/master/Maintainers.md) it says 
> > while reviewing PRs, leave your comment even if everything looks good - simple "Looks good to me" or "LGTM" will suffice, so that we know someone has already taken a look at it
> 
> But in this text says you should resolve the comments. These texts need to be "aligned". :)
> 
> 
</details>
Yup. Now that this document includes more detailed instructions on reviewing PRs, I can point that document to this one. The maintainers doc actually needs some more updates since I also updated all the labels so I would prefer it to be in a separate PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 24 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/182#-:-KR5sWwm6c7f4IH-AO6A:bnfp4nl)*
<!-- Sent from Reviewable.io -->
